### PR TITLE
added list command filter by side

### PR DESCRIPTION
**Pull request by JaegyuDev**
This is my first OS PR!
This should close issue #245 

I was looking for previous examples of this in the code base and I saw examples of this in `curseforge/export.go`.
My one concern is the use of a flag for the side. Is it alright in this case or should I change something?

### CODE REVIEW

File: cmd/list.go [hunk 1]
```diff
@@ -39,6 +39,24 @@ var listCmd = &cobra.Command{
 			os.Exit(1)
 		}
 
+		// Filter mods by side
+		if viper.IsSet("list.side") {
+			side := viper.GetString("list.side")
+			if len(side) == 0 || (side != core.UniversalSide && side != core.ServerSide && side != core.ClientSide) {
+				fmt.Println("Invalid side!")
```
Comment (comp500) [inline cmd/list.go:46]: Could you list the possible options the user has? e.g. `Invalid side "front", must be one of client, server, or both (default)`

**Review by comp500 — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks good, just had a suggestion to improve the UX a bit.

**Comment by 3AM-Developer** [comment]
Should I include this change to the Export command for Curseforge?

Currently it looks like this:
```
side := viper.GetString("curseforge.export.side")
if len(side) == 0 || (side != core.UniversalSide && side != core.ServerSide && side != core.ClientSide) {
	fmt.Println("Invalid side!")
	os.Exit(1)
}
```

File: cmd/list.go [hunk 2]
```diff
@@ -39,6 +39,25 @@ var listCmd = &cobra.Command{
...
+			// cobra/viper ensures this value is set if the flag is used. Maybe remove the len Check?
+			side := viper.GetString("list.side")
+			if side != core.UniversalSide && side != core.ServerSide && side != core.ClientSide {
+				fmt.Printf("Invalid side %q, must be one of client, server, or both (default)\n", side)
+				os.Exit(1)
+			}
+
+			i := 0
+			for _, mod := range mods {
+				if len(side) == 0 || (mod.Side == side || mod.Side == core.EmptySide || mod.Side == core.UniversalSide || side == core.UniversalSide) {
```
Comment (comp500) [inline cmd/list.go:53]: No need for the len check here - the previous if already validates that the side is one of both/server/client

File: curseforge/export.go [hunk 3]
```diff
@@ -19,9 +20,10 @@ var exportCmd = &cobra.Command{
 	Short: "Export the current modpack into a .zip for curseforge",
 	Args:  cobra.NoArgs,
 	Run: func(cmd *cobra.Command, args []string) {
+		// cobra/viper ensures this value is set if the flag is used. Maybe remove the len Check?
```
Comment (comp500) [inline curseforge/export.go:23]: Yeah you can remove the len check here as well - the latter checks already handle the case where the string is empty, and it's very unlikely that a user would manage to set the flag to `""`

**Comment by JaegyuDev** [comment]
Should be good to go I believe!

**Comment by JaegyuDev** [comment]
Quick question: in the future if I push code that has a minor issue like leaving a comment in, should I do 2 commits like I just did, or is there a better/preferred way of fixing something like that?

**Comment by comp500** [comment]
Making 2 commits is usually easiest, you can amend commits but if you've already pushed you'd need to do a force push to overwrite the existing commit. I usually squash PRs into a single commit so it doesn't make a difference once merged.

**Review by comp500 — APPROVED** [review]

**Pull request merged**